is_core/main.py

```diff
@@ -490,30 +483,37 @@ def get_rest_form_exclude(self, request, obj=None):
         return self.get_form_exclude(request, obj)
 
     def get_rest_extra_fields(self, request, obj=None):
-        if self.rest_extra_fields:
-            return rfs(self.rest_extra_fields)
-        else:
-            return rfs(self.model._rest_meta.extra_fields).join(rfs(self.rest_default_extra_fields))
-
-    def get_rest_general_fields(self, request, obj=None):
-        if self.rest_general_fields:
-            return rfs(self.rest_general_fields)
-        else:
-            return rfs(self.model._rest_meta.default_general_fields).join(rfs(self.rest_default_general_fields))
-
-    def get_rest_detailed_fields(self, request, obj=None):
-        if self.rest_detailed_fields:
-            return rfs(self.rest_detailed_fields)
-        else:
-            return rfs(self.model._rest_meta.default_detailed_fields).join(rfs(self.rest_default_detailed_fields))
+        return list(
+            self.model._rest_meta.extra_fields if self.rest_extra_fields is None
+            else self.rest_extra_fields
+        )
 
-    def get_rest_guest_fields(self, request, obj=None):
-        if self.rest_guest_fields:
-            return rfs(self.rest_guest_fields)
-        else:
-            return rfs(self.model._rest_meta.guest_fields).join(rfs(self.rest_default_guest_fields))
+    def get_rest_default_fields(self, request, obj=None):
+        return list(
+            self.model._rest_meta.default_fields if self.rest_default_fields is None
+            else self.rest_default_fields
+        )
+
+    def get_rest_fields(self, request, obj=None):
+        return self.rest_fields
 
-        return rfs(self.rest_guest_fields)
+    def get_rest_general_fields(self, request):
+        return list(
+            self.model._rest_meta.general_fields if self.rest_general_fields is None
+            else self.rest_general_fields
+        )
```

Comment:
všechno je to list schválně z toho vytvářím nové listy aby se nestal nějaký problém se změnou statické proměnné